v2/server.go

```diff
@@ -94,48 +92,124 @@ type Server struct {
 	statusReceiveOverrideFn func(e cloudevents.Event, dataChan *channel.DataChan) error
 }
 
-// publisher/subscription data model
+// SubscriptionInfo
+//
+// SubscriptionInfo defines data types used for subscription.
+// swagger:parameters
+type SubscriptionInfo struct { //nolint:deadcode,unused
+	// Identifier for the created subscription resource.
+	// example: d1dd1770-e718-401e-ba32-cef05a286164
+	// +required
+	ID string `json:"SubscriptionId" omit:"empty"`
+	// Endpoint URI (a.k.a callback URI), e.g. http://localhost:8080/resourcestatus/ptp
+	// example: http://consumer-events-subscription-service.cloud-events.svc.cluster.local:9043/event
+	// +required
+	EndPointURI string `json:"EndpointUri" example:"http://localhost:8080/resourcestatus/ptp" omit:"empty"`
+	// The URI location for querying the subscription created.
+	// example: http://localhost:9043/api/ocloudNotifications/v2/publishers/d1dd1770-e718-401e-ba32-cef05a286164
+	// +required
+	URILocation string `json:"UriLocation" omit:"empty"`
+	// The resource address specifies the Event Producer with a hierarchical path.
+	// Format /{clusterName}/{siteName}(/optional/hierarchy/..)/{nodeName}/{(/optional/hierarchy)/resource}
+	// example: /east-edge-10/vdu3/o-ran-sync/sync-group/sync-status/sync-state
+	// +required
+	Resource string `json:"ResourceAddress" example:"/east-edge-10/vdu3/o-ran-sync/sync-group/sync-status/sync-state"`
+}
+
+// Event Data Model
+//
+// Event Data Model specifies the event Status Notification data model supported by the API. The current model supports JSON encoding of the CloudEvents.io specification for the event payload.
+// swagger:parameters
+type EventData struct {
```

Comment:
Same here?

Reply:
Because of the way cloudevents code structured, it is impossible to generate a clear viewable schema.

See here:
https://pkg.go.dev/github.com/cloudevents/sdk-go/v2@v2.15.2/event#EventContextV1

Notice there is no `Data` field but an `Extensions` field with interface{} to plugin general data.

I have tried many ways to use data models from cloudevents library directly but it is just not possible.